### DISCUSSION TITLE
Compiling for MAC or linux

### DISCUSSION THREAD

**Discussion by SashaMilic** [nofeletru/UsbAsp-flash#55] [Q&A] [answered]
Hi,

First I wish to tell you that you are great..!!! And that you help so much people for flashing they bios and get out of situation with bricked motherboards.

Considering I'm a mac user it will be great if you can help me to port AsProgrammer to macOS and maybe it can be ported on linux.

Can you post procedure how you compile and get exec file, I will be glad to help and barrow my mac to compile it for macOS.

Thanks in advance,
Sasha Milic

**Comment by nofeletru** [comment, marked as answer]
Unfortunately the software uses windows libs for CH341a and FT232H, so you can't just compile it for unix. Therefore you need rewrite those units for libusb. The easy way just use virtual machine.
